onebusaway-android/src/main/java/com/joulespersecond/oba/ObaAnalytics.java

```diff
@@ -0,0 +1,260 @@
+
+    /**
+     * To measure the distance when the bus stop tapped.
+     */
+    public enum ObaStopDistance {
+
+        DISTANCE_1("User Distance: 00000-00050m", 50),
+        DISTANCE_2("User Distance: 00050-00100m", 100),
+        DISTANCE_3("User Distance: 00100-00200m", 200),
+        DISTANCE_4("User Distance: 00200-00400m", 400),
+        DISTANCE_5("User Distance: 00400-00800m", 800),
+        DISTANCE_6("User Distance: 00800-01600m", 1600),
+        DISTANCE_7("User Distance: 01600-03200m", 3200),
+        DISTANCE_8("User Distance: 03200-∞m", 0);
+
+        private final String stringValue;
+        private final int distanceInMeters;
+
+        private ObaStopDistance(final String s, final int i) {
+            stringValue = s;
+            distanceInMeters = i;
+        }
+
+        public String toString() {
+            return stringValue;
+        }
+
+        public int getDistanceInMeters() {
+            return distanceInMeters;
+        }
+    }
+
+    /**
+     * Event categories for segmentation
+     * app_settings, ui_action, submit is similar with OBA IOS
+     */
+    public enum ObaEventCategory {
+
+        APP_SETTINGS("app_settings"), UI_ACTION("ui_action"),
+        SUBMIT("submit"), STOP_ACTION("stop_metrics"),
+        ACCESSIBILITY("accessibility");
+
+        private final String stringValue;
+
+        private ObaEventCategory(final String s) {
+            stringValue = s;
+        }
+
+        public String toString() {
+            return stringValue;
+        }
+    }
+
+    /**
+     * Reports events with categories. Helps segmentation in GA admin console.
+     *
+     * @param category category name
+     * @param action   action name
+     * @param label    label name
+     */
+    public static void reportEventWithCategory(String category, String action, String label) {
+        Tracker tracker = Application.get().getTracker(Application.TrackerName.APP_TRACKER);
+        Tracker tracker2 = Application.get().getTracker(Application.TrackerName.GLOBAL_TRACKER);
+        String obaRegionName = getObaRegionName();
+
+        tracker.send(new HitBuilders.EventBuilder()
+                .setCategory(category)
+                .setAction(action)
+                .setLabel(label)
+                .setCustomDimension(1, obaRegionName)
+                .build());
+
+        tracker2.send(new HitBuilders.EventBuilder()
+                .setCategory(category)
+                .setAction(action)
+                .setLabel(label)
+                .setCustomDimension(1, obaRegionName)
+                .build());
+    }
+
+    /**
+     * Tracks stop tap distance between bus stop location and users current location
+     *
+     * @param stopId       for action
+     * @param myLocation   the users location
+     * @param stopLocation tapped stop location
+     */
+    public static void trackBusStopDistance(String stopId, Location myLocation, Location stopLocation) {
+        try {
```

Comment:
Let's remove the try/catch block, and just add at the top of the method:

~~~
if(myLocation == null) {
    return;
}
~~~

Its a little cleaner - generally exceptions should be exceptional cases, and if we know `myLocation` can be null, we should handle it in code rather than throwing an exception.